core/headerchain.go

```diff
@@ -501,6 +501,57 @@ func (hc *HeaderChain) findCommonAncestor(header *types.WorkObject) *types.WorkO
 	}
 
 }
+func (hc *HeaderChain) WorkShareDistance(wo *types.WorkObject, ws *types.WorkObjectHeader) (*big.Int, error) {
+	current := wo
+	// Create a list of ancestor blocks to the work object
+	ancestors := []*types.WorkObject{}
```

Comment:
This process would be more efficient if ancestors was a map. Then checkInAncestorsList would disappear too.

Reply:
changed it